SEIRD vignette following Lisa's comments (#38)
Como-DTC-Collaboration/como-models: vignettes/SEIRD.Rmd

Note that $\beta$ is a rate (per unit time) at which a susceptible individual becomes infected on contact with an infected individual. It is not a probability as it is sometimes mistakenly labelled and so $\beta > 1$ is possible.

> The full SEIRD model consists of four ODEs that a coupled together: the first of these governs the behaviour of susceptible individuals, which has already been introduced. The next equation governs the growth of the exposed compartment:

Reviewer: ```suggestion
The full SEIRD model consists of four ODEs that are coupled together: the first of these governs the behaviour of susceptible individuals, which has already been introduced. The next equation governs the growth of the exposed compartment:
```

$$E(t) = E_0 \exp (- \kappa t)$$
representing exponential decay, which we can plot assuming $E_0=1$ and $\kappa=0.5$:
```{r, fig.height = 5, fig.width = 7}
# function to return exponential decay
E <- function(E_0, kappa, t) {E_0 * exp(-kappa * t)}

# make values to plot
E_0 <- 1
kappa <- 0.5
t <- seq(0, 10, 0.01)
E_vals <- E(E_0, kappa, t)

tibble(t=t, E=E_vals) %>% 
  ggplot(aes(x=t, y=E)) +
  geom_line() +
  xlab("time (days)")
```

We can also show the effect of changing $\kappa$: unsurprisingly, as $\kappa$ increases, the rate at which individuals become infectious increases and there is faster decay.
```{r, fig.height = 5, fig.width = 7}
kappa_1 <- 0.5
kappa_2 <- 1
kappa_3 <- 2
E_1 <- E(E_0, kappa=kappa_1, t=t)
E_2 <- E(E_0, kappa=kappa_2, t=t)
E_3 <- E(E_0, kappa=kappa_3, t=t)
tibble(t, E=E_1, kappa=kappa_1) %>% 
  bind_rows(tibble(t, E=E_2, kappa=kappa_2)) %>% 
  bind_rows(tibble(t, E=E_3, kappa=kappa_3)) %>% 
  ggplot(aes(x=t, y=E, colour=as.factor(kappa))) +
  geom_line() +
  scale_color_brewer("kappa", palette = "Dark2") +
  xlab("time (days)")
```

> An alternative way to view these exponential decay curves is as probability distributions governing the time taken for an exposed individual to become infectious. This period is referred to as the latency period. Whilst rigorous proof of this requires considering the equivalent stochastic version of this model, the mean from the deterministic model matches the mean of the stochastic equivalent. As such, the waiting time for an individual to transition from exposed to infectious classes is governed by an exponential distribution. Using repeatedly sampled PCR test data from close contacts of confirmed cases in mainland China, a recent study estimated the latency period of COVID-19 as 5.51 days and a log-normal probability distribution was found to represent that waiting period (Xi et al., Estimating the Latent Period of Coronavirus Disease 2019, 2021). Below we compare the waiting period for the log-normal distribution fit in this study with an exponential distribution with the same latency period.

Reviewer: I think this paragraph needs 1--2 sentences extra explanation---some readers may be confused that we are suddenly talking about probability distributions and a mean in the deterministic model.

Reviewer 2: Thanks Richard. I've changed this now.

$$\frac{\text{d}C}{\text{d}t} = \text{newly exposed} = \beta S I,$$
where we assume $C(0) = 0$. The number of new infections arising between two time intervals $t_1$ and $t_2$ is then: $C(t_2) - C(t_1)$. Similarly, the number of deaths occuring over the same interval is the difference in the cumulative deaths: $D(t_2) - D(t_1)$.

## Steady states
We can now consider whether there are equilibrium states of the system: values of the state variables where, unless perturbed, the system will remain where it is. To do so, we set the right-hand sides of the ODEs to zero. Taking the equation for the 
susceptibles:

$$\frac{\text{d}S}{\text{d}t} = -\beta S I = 0.$$

This only has solutions if either $S=0$, in which case there are no individuals left to infect; or $I=0$, in which case there are no infectious individuals: in other words, an infection has not been imported into the population. Considering all four ODEs, $I=0$ implies that $\frac{\text{d}R}{\text{d}t} = 0$ and $\frac{\text{d}D}{\text{d}t} = 0$. Additionally, so long as there are no exposed individuals, $\frac{\text{d}E}{\text{d}t}= 0$ and $\frac{\text{d}I}{\text{d}t}= 0$ and 

> So the equilbria are given by $S=S'$, $E=0$, and $I=0$ which corresponds to a situation when there are no infected or infectious individuals in the population. In other words, in this model, there is no endemic equilbrium when the infection remains present throughout the population; only epidemic dynamics are possible.

Reviewer: ```suggestion
So the equilibria are given by $S=S'$, $E=0$, and $I=0$ which corresponds to a situation when there are no infected or infectious individuals in the population. In other words, in this model, there is no endemic equilibrium when the infection remains present throughout the population; only epidemic dynamics are possible.
```

  else
    result <- result %>% bind_rows(temp)
}

# plot results
result %>% 
  filter(compartment=="I") %>% 
  ggplot(aes(x=time, y=value, colour = beta, group=as.factor(beta))) +
  geom_line() +
  scale_color_viridis_c() +
  xlab("time (days)") +
  ylab("Infectious proportion")
```

Why do some of the simulations produce epidemic waves whereas others fail to do so?

> This is can be explored through the basic reproductive number, $R_0$, which is the expected number of people infected by a single infectious individual in a population where everyone is susceptible. (Note that this basic reproductive numberis unrelated to the initial condition for the recovered population.) 

Reviewer: ```suggestion
This is can be explored through the basic reproductive number, $R_0$, which is the expected number of people infected by a single infectious individual in a population where everyone is susceptible. (Note that this basic reproductive number is unrelated to the initial condition for the recovered population.) 
```
